internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+	)
+	if err != nil {
+		res.Diagnostics.AddError(
+			"Error deleting business unit",
+			"Could not delete business unit, unexpected error: "+err.Error(),
+		)
+		return
+	}
+}
+
+// Read implements resource.Resource.
+func (b *companyResource) Read(ctx context.Context, req resource.ReadRequest, res *resource.ReadResponse) {
+	var state Company
+	diags := req.State.Get(ctx, &state)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	company, err := b.client.BusinessUnits().WithId(state.ID.ValueString()).Get().Execute(ctx)
+	if err != nil {
+		if errors.Is(err, platform.ErrNotFound) {
+			res.State.RemoveResource(ctx)
+			return
+		}
+
+		res.Diagnostics.AddError(
+			"Error reading business unit",
+			"Could not retrieve the business unit, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	current := NewCompanyFromNative(company)
+
+	diags = res.State.Set(ctx, current)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update implements resource.Resource.
+func (b *companyResource) Update(ctx context.Context, req resource.UpdateRequest, res *resource.UpdateResponse) {
+	var plan Company
+	diags := req.Plan.Get(ctx, &plan)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	var state Company
+	diags = req.State.Get(ctx, &state)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	input := state.updateActions(plan)
+	var company *platform.BusinessUnit
+
+	err := retry.RetryContext(ctx, 5*time.Second, func() *retry.RetryError {
+		var err error
+		company, err = b.client.BusinessUnits().
+			WithId(state.ID.ValueString()).
+			Post(input).
+			Execute(ctx)
+
+		return utils.ProcessRemoteError(err)
+	})
+	if err != nil {
+		res.Diagnostics.AddError(
+			"Error updating business unit",
+			"Could not update business unit, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	current := NewCompanyFromNative(company)
+	diags = res.State.Set(ctx, &current)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+}
+
+func NewCompanyResource() resource.Resource {
+	return &companyResource{}
+}
+
+var (
+	_ resource.Resource                = &divisionResource{}
+	_ resource.ResourceWithConfigure   = &divisionResource{}
+	_ resource.ResourceWithImportState = &divisionResource{}
+)
+
+type divisionResource struct {
+	client *platform.ByProjectKeyRequestBuilder
+}
+
+// Schema implements resource.Resource.
+func (b *divisionResource) Schema(_ context.Context, req resource.SchemaRequest, res *resource.SchemaResponse) {
+	res.Schema = schema.Schema{
+		Description: "Business Unit type to represent the top level of a business. Contains specific fields and values that differentiate a Company from the generic BusinessUnit.\n\n" +
+			"See also the [Business Unit API Documentation](https://docs.commercetools.com/api/projects/business-units",
+		Attributes: map[string]schema.Attribute{
```

Comment:
storeMode is missing as an option (and can be optional and default to `FromParent`